src/types/set.rs

```diff
@@ -71,96 +71,248 @@ impl PySet {
     where
         K: ToPyObject,
     {
-        fn inner(set: &PySet, key: PyObject) -> PyResult<bool> {
+        Bound::borrowed_from_gil_ref(&self).contains(key)
+    }
+
+    /// Removes the element from the set if it is present.
+    ///
+    /// Returns `true` if the element was present in the set.
+    pub fn discard<K>(&self, key: K) -> PyResult<bool>
+    where
+        K: ToPyObject,
+    {
+        Bound::borrowed_from_gil_ref(&self).discard(key)
+    }
+
+    /// Adds an element to the set.
+    pub fn add<K>(&self, key: K) -> PyResult<()>
+    where
+        K: ToPyObject,
+    {
+        Bound::borrowed_from_gil_ref(&self).add(key)
+    }
+
+    /// Removes and returns an arbitrary element from the set.
+    pub fn pop(&self) -> Option<PyObject> {
+        Bound::borrowed_from_gil_ref(&self).pop().map(|e| e.into())
+    }
+
+    /// Returns an iterator of values in this set.
+    ///
+    /// # Panics
+    ///
+    /// If PyO3 detects that the set is mutated during iteration, it will panic.
+    pub fn iter(&self) -> PySetIterator<'_> {
+        IntoIterator::into_iter(self)
```

Comment:
As with frozenset, I would be glad if we could standardize all iterator related methods to be implemented in terms of the `BoundSetIterator::new` constructor.